### ISSUE TITLE
Ubuntu 18.04 using Wayland

### ISSUE THREAD

**Issue by frohro** [andyholmes/gnome-shell-extension-gsconnect#99]
I may have messed things up when I had both kdeconnect and gsconnect both installed simultaneously, but it almost behaved better then than now.  I am using v11 from git.  I have installed all the packages listed as dependencies plus the caribou one.  This is a fresh install of Ubuntu 18.04, so it is probably missing something.  The problems I have compared to what I see on the photo on the main github page are:

- Since I uninstalled kdeconnect, I have not seen my phone listed under the menu on the right, under Mobile Devices heading as shown in the screenshot.  There is only the Mobile Settings option.  I have seen it a few times under a phone icon on the top panel, but it isn't there sometimes too.  I actually saw the menu behave as I the screenshot shows early in my testing, but I also had an entry for the computer that I believe was because I had kdeconnect also installed.

- There is no menu option for gsconnect under the right click nautilus menu as shown in the screenshot.

- The remote mouse doesn't work, as reported in another bug below.  I added a comment to that bug report.

There are some things that do work:  They are ping, sending a file from the phone to the laptop, remote commands, remote media control, pairing and unpairing, and notifications.

I'm curious where the settings are retained.  I noticed that at first I had to enable features in the preferences, but now I don't, even though I uninstall it using the option on the browser gnome extensions  page.  It doesn't seem to remove the settings when it uninstalls.  I wonder if I could delete them, if things would be fixed.

Thanks for your good work.  I'm happy to try and help debug these things if I can be of any use to you.
Rob

**Comment by andyholmes** [comment]
> Since I uninstalled kdeconnect, I have not seen my phone listed under the menu on the right...

If you had KDE Connect installed when GSConnect was first started then it likely started on port 1717, which will make it difficult for the Android app to find. The KDE Connect server may continue running after uninstalled, though. You should kill that process and also kill GSConnect, which should restart automatically.

    killall -9 kdeconnectd
    kill -9 $(ps aux | grep '[d]aemon.js' | awk '{print $2}')

> There is no menu option for gsconnect under the right click nautilus menu as shown in the screenshot.

For Nautilus integration to work both the *Share* plugin and *Nautilus Integration* must be enabled in Preferences and Nautilus must be restarted (`nautilus -q`). Additionally, it will only be visible when at least one device with the *Share* plugin enabled is paired and connected.

> The remote mouse doesn't work, as reported in another bug below. I added a comment to that bug report.

I saw that also. [Remote Input](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Frequently-Asked-Questions#why-cant-i-use-remote-input-in-wayland) is not available under Wayland unfortunately.

> I'm curious where the settings are retained.

Most [settings are in GSettings/dconf](https://github.com/andyholmes/gnome-shell-extension-gsconnect/wiki/Frequently-Asked-Questions#where-are-the-settings-stored), except for the local certificate in `~/.config/gsconnect` and the cache in `~/.cache/gsconnect`. Deleting settings might help if discovery fails, but generally it shouldn't be necessary.

**Comment by frohro** [comment]
Thanks Andy,
You solved several of my issues; deleting the settings with dconf produced these results: the device is now showing up in the drop down menu instead of directly on the menu bar, there is now a warning when I try to control the mouse pointer, and I understood that Wayland is not set up for controlling the pointer, so that is good.
I still cannot see any option in nautilus for saving to the mobile device, though I can share files both ways using the share file/url or browse files, and from the phone, send files.

One other annoying thing I can't seem to fix is that KDEconnect disconnects from the computer whenever the screen turns off.  I did some testing, and found that my phone is sleeping the WIFI for about 20 seconds, then waking it up again, for a few seconds, and then sleeping it again.  I would like the GSconnect not to disconnect for that kind of behavior.  Is there a place I can change a timeout for that?
Thanks again for the great help!
Rob

**Comment by andyholmes** [comment]
> I still cannot see any option in nautilus for saving to the mobile device, though I can share files both ways using the share file/url or browse files, and from the phone, send files.

Have you enabled *Nautilus Integration* in the preferences dialog and restarted Nautilus (`nautilus -q`)?

> I would like the GSconnect not to disconnect for that kind of behavior. Is there a place I can change a timeout for that?

This isn't anything GSConnect can deal with unfortunately; it's up to the phone whether it decides to close connections when it goes to sleep. The KDE Connect protocol uses a 30 second keepalive for TCP connections so if disconnects are happening under that time then that must be an Android issue.

**Comment by andyholmes** [comment]
It's been two weeks since any activity, is this issue still valid?

**Comment by andyholmes** [comment]
Closing due to lack of activity, re-open if any of the issues persist.

**Comment by antoniusblock666** [comment]
> Have you enabled Nautilus Integration in the preferences dialog and restarted Nautilus (nautilus -q)?

I've been having this issue since I upgraded from Ubuntu 17.10 to 18.04 (it was working fine on 17.10). I've checked the preferences and plugins as @andyholmes suggested but no luck. I can share files without any trouble from the share files option in GSConnect system indicator in the Ubuntu top bar. But when I right click on a file in Nautilus it doesn't show any option to send to my device.

Any suggestions on what I could try?
